bob/src/cluster/quorum.rs

```diff
@@ -272,21 +272,130 @@ impl Cluster for Quorum {
     }
 
     async fn exist(&self, keys: &[BobKey]) -> Result<Vec<bool>, Error> {
-        let keys_by_nodes = group_keys_by_nodes(&self.mapper, keys);
-        debug!(
-            "EXIST Nodes for fan out: {:?}",
-            &keys_by_nodes.keys().flatten().collect::<Vec<_>>()
-        );
         let len = keys.len();
         let mut exist = vec![false; len];
-        for (nodes, (keys, indexes)) in keys_by_nodes {
-            let cluster_results = LinkManager::exist_on_nodes(&nodes, &keys).await;
-            for result in cluster_results.into_iter().flatten() {
-                for (&r, &ind) in result.inner().iter().zip(&indexes) {
-                    exist[ind] |= r;
+
+        // filter local keys
+        let mut indices = Vec::new();
+        let local_keys = keys
+            .iter()
+            .enumerate()
+            .filter_map(|(idx, &key)| {
+                if self
```

Comment:
Maybe it's better to use `find(...).map(|idx| {indices.push(idx); idx})` to remove `if` and explicit `Some` / `None`

Reply:
I was able to eliminate `is_some` call using `find_map` in [2695039](https://github.com/qoollo/bob/pull/634/commits/2695039d6d39b4b0353ae0b3f62bfe70f8c33091).

Using `find(...).map(...)` on `keys` will not work as `find` will return the first key and it is needed to iterate over every key.